src/sinks/file/mod.rs

```diff
@@ -134,7 +179,7 @@ impl FileSink {
                         Some(Ok((mut expired_file, path))) => {
                             // We got an expired file. All we really want is to
                             // flush and close it.
-                            if let Err(error) = expired_file.flush().await {
+                            if let Err(error) = expired_file.shutdown().await {
```

Comment:
Yeah, I had written some test code with async_compression library and just using `flush`, the gzip file was not being written properly. Using shutdown, however, was closing the file properly. I had guessed that using shutdown for GzipEncoder flushes with required gzip headers, which are not proper with just `flush`.
I havn't dug in async_compression code so not sure.

Reply:
FTR a `flush` should just ensure whatever is buffered is written out to the compressed stream, but that won't necessarily ensure that the file is closed off properly, which involves fixing up the header and checksums. The latter should only ever be done once, hence a separate `shutdown`.

Reply:
I just realize that `shutdown` requires to be called on all files, especially in case of `GzipEncoder` when vector shuts down. Else the required gzip headers won't be written. 

I am not sure how/where this shutdown handling will be added.

Reply:
Also, I went through `tokio` code and, for `File` `shutdown` won't flush the buffers, it just returns `Ok`. This is a bug since it violates `AsyncWrite` contract. https://github.com/tokio-rs/tokio/issues/2760 

We should call `flush` and `shutdown` until that issue is resolved, and link the issue in the comment.

Reply:
Nope disregard my comment before, all is fine.